### PR TITLE
feature(DropdownGroup.container): Displaying the size and quantity ba…

### PR DESCRIPTION
**Pull request by ShravyaGurmitkal**
…sed on selected color

# Description:

Displaying the size and quantity to only specific range for the selected color.
Also changed the swagger comments of variants endpoint.

# Jira:

https://hackyourfuture-dk.atlassian.net/browse/CLASS22-150

### CODE REVIEW

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js [hunk 1]
```diff
@@ -33,16 +36,56 @@ export const DropdownGroup = ({ productId }) => {
   sizes = [...new Set(sizes)];
   quantities = [...new Set(quantities)];
 
+  const fetchVariantsByColor = async (color) => {
+    let filteredSizes = [];
+    let filteredQuantities = [];
+    const response = await fetch(
+      `${apiURL()}/variants?color=${color}&product_id=${productId}`,
```
Comment (orhantoy) [inline packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js:45]: I would prefer that the endpoint URL is like this: `/api/products/:productId/variants`
The color can still be a query param, that's OK.

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js [hunk 2]
```diff
@@ -33,16 +36,56 @@ export const DropdownGroup = ({ productId }) => {
...
+    const filteredVariantsData = await response.json();
+    filteredVariantsData.forEach((variant) => {
+      filteredSizes.push(variant.size);
+      filteredQuantities.push(variant.stock);
+    });
+    filteredSizes = [...new Set(filteredSizes)];
+    filteredQuantities = [...new Set(filteredQuantities)];
+    const maxQuantity = Math.max(...filteredQuantities);
+    if (maxQuantity >= 5) {
+      filteredQuantities = [...Array(10).keys()].map((i) => i + 1);
```
Comment (orhantoy) [inline packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js]: What happens here if `maxQuantity` is 7?

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js (hunk 1 above)
Comment (ShravyaGurmitkal) [reply packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js:43]: But when we want to find variants using variant id then we will use path parameter, its using productId so I have used it as query parameter

Comment (ShravyaGurmitkal) [reply packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js:45]: Now should we need to create new endpoint

Comment (ShravyaGurmitkal) [reply packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js:45]: I have used the endpoint which was already created for this task

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js (hunk 2 above)
Comment (ShravyaGurmitkal) [reply packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js]: Yeah I will change this

**Review by viedamayanti — APPROVED** [review]

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js (hunk 1 above)
Comment (orhantoy) [reply packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js:45]: Ok, it's fine to keep as is then.

**Review by orhantoy — APPROVED** [review]
I haven't reviewed in detail so I'm just approving to unblock you.

**Pull request merged**